lib/Net/HTTP.pm

```diff
@@ -3,7 +3,7 @@ our $VERSION = '6.20';
 use strict;
 use warnings;
 
-use vars qw($SOCKET_CLASS);
+our $SOCKET_CLASS;
 unless ($SOCKET_CLASS) {
```

Comment:
These lines will no longer do what they used to, because `use vars` is somewhat different from `our` in the way it does namespacing.

With this new code, $SOCKET_CLASS will always be undefined in the lines here that check it, so we might as well just remove all that code -- but we still do want code that supports different backends.

Reply:
So I'm trying to understand what you mean and write tests, but I can't reproduce what you state, this seems to DTRT to me:
```perl
$Net::HTTP::SOCKET_CLASS = 'Foo';

require Net::HTTP;

is $Net::HTTP::SOCKET_CLASS, 'Foo';

is_deeply \@Net::HTTP::ISA, [qw[Foo Net::HTTP::Methods]];
```